hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java

```diff
@@ -287,9 +283,12 @@ public Option<BufferedRecord<T>> deltaMergeNonDeleteRecord(BufferedRecord<T> new
         if (combinedRecordData != existingRecord.getRecord()) {
           Pair<HoodieRecord, Schema> combinedRecordAndSchema = combinedRecordAndSchemaOpt.get();
           return Option.of(BufferedRecord.forRecordWithContext(combinedRecordData, combinedRecordAndSchema.getRight(), readerContext, orderingFieldName, false));
+        } else {
+          return Option.empty();
         }
       }
-      return Option.empty();
+      // An empty Option indicates that the output represents a delete.
+      return Option.of(new BufferedRecord<>(newRecord.getRecordKey(), newRecord.getOrderingValue(), null, null, true));
```

Comment:
Similarly here, this empty option is translated to a delete [here](https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java#L309)

Reply:
Is it correct to use `newRecord.getOrderingValue()`, or should we use `DEFAULT_ORDERING_VALUE` (0)?